### DISCUSSION TITLE
Option to read GE-Prorton changes before download

### DISCUSSION THREAD

**Discussion by SaschaVasarrhelyi** [DavidoTek/ProtonUp-Qt#367] [Ideas]
It is possible to make a Funktion to read the changes list for an Proton-GE to take the right Proton-GE Version for a specific game?

**Comment by sonic2kk** [comment]
Do you mean download a specific version of GE-Proton that may include fixes for a specific game? If so, this shouldn't be necessary; just download the latest.

Apart from that, I don't think there's a good way to do this. Reading the changelog and trying to infer this kind of information is probably not a good idea. I don't think it could be done reliably

**Comment by SaschaVasarrhelyi** [comment]
Yes, i mean an changelog for an specific proton-ge, then it is not secur that the newest proton-ge have all the patches for a specific game that needs an older version of proton-ge to start.
But if it's impossible it's okay to. 
ProtonUp-QT have so on so not all Proton-ge to download but just a small list of various protn-ge- version at the list.
But thanks to think about it.

**Comment by sonic2kk** [comment]
Others are free to share their opinions too, but from experience *I* feel like trying to parse this sort of thing from text is not really a reliable way of doing things.

I am just a user of and occasional contributor to ProtonUp-Qt by the way, my word is not final, so hope you didn't feel forced to close your discussion!

**Comment by DavidoTek** [comment]
I like the idea, but I don't think that would work.

> from experience I feel like trying to parse this sort of thing from text is not really a reliable way of doing things.

Yes, manually implementing a parser is not viable as there is no specification for the changelog and it would not be easy to understand the semantics.
I guess in the era of AI and Chat-GPT this isn't "impossible", but that also wouldn't be reliable at all because it also would need to consider reports from protondb.com etc. In addition to that, a lot of cloud interaction would be required. I don't think we want to do that.
